### ISSUE TITLE
New description scale for Concentration Level

### ISSUE THREAD

**Issue by terracoda** [phetsims/molarity#117]
Related to #115, #116 the described qualitative scale for Concentration Levels changed from a 6-point scale to a 7-point scale. Concentration level is solute dependent, so now values provided here or in design doc.

@Matthew-phet please assign to @twant when you are ready.
Note that the verb is part of the parameter because it changes dependent on the description value. Please handle this as you see fit.

Old Descriptions | New Descriptions
-- | --
{{max concentration}} | {{has max concentration}}
{{very concentrated}} | {{is highly concentrated}}
{{concentrated}} | {{is very concentrated}}
{{slightly concentrated}} | {{is not very concentrated}}
{{barely concentrated}} | {{is slightly concentrated}}
{{not concentrated}} | {{has low concentration}}
  | {{has zero concentration}}

**Comment by Matthew-phet** [comment]
looks good to me! assigning to @twant

**Comment by twant** [comment]
@Matthew-phet what numerical values would you like assigned to these descriptions?

**Comment by terracoda** [comment]
@Matthew-phet, my understanding is that concentration varies for each solute. 
Would the following make sense:
-Make "has max concentration" the value of the solute's saturation point (and anything beyond it)
- Make "has zero concentration" a single value when there is no solute
- Make the rest of the old descriptions map directly over to the remaining new descriptions like so:
  - {{very concentrated}} = {{is highly concentrated}}
  - {{concentrated}} = {{is very concentrated}}
  - {{slightly concentrated}} = {{is not very concentrated}}
  - {{barely concentrated}} = {{is slightly concentrated}}
  - {{not concentrated}} = {{has low concentration}}

So what ever was the range for "very concentrated" before will now be "is highly concentrated".
@Matthew-phet, would that make sense?

**Comment by Matthew-phet** [comment]
Yup! that was my understanding as well.

**Comment by terracoda** [comment]
@twant, do you have any further questions on the Concentration Levels? I did not see the new concentration levels implemented on Master when I tested the sim today. Here are a couple of screen shots:
<img width="930" alt="concentration-level-screen-summary-play-area" src="https://user-images.githubusercontent.com/1902116/60506737-cb503600-9cc6-11e9-9320-53be0450b15d.png">

<img width="932" alt="Screen Shot 2019-07-02 at 12 43 07 PM" src="https://user-images.githubusercontent.com/1902116/60507074-9abccc00-9cc7-11e9-9f6e-7ed28fa9c9fa.png">

**Comment by terracoda** [comment]
Also, there is special wording for when there is "no solute" and solution is actually "pure water".

**Comment by terracoda** [comment]
@twant, apologies, the screen summary wording including special cases is covered in #123. We can just stick to implementation of "concentration levels" in this issue.

**Comment by terracoda** [comment]
@twant, the description scale that describes concentration level will also be included in the alerts (i.e., context response) that fire for a newly chosen Solute (#127). 

In the alerts, however, there is no verb. I just wanted to mention that as I am not sure how the wording in the larger phrases and the wording in the actual parameters are being managed.

**Comment by twant** [comment]
@terracoda @Matthew-phet thanks for your patience with this one! assigning back over to you.

**Comment by terracoda** [comment]
<img width="856" alt="Screen Shot 2019-07-23 at 9 14 11 AM" src="https://user-images.githubusercontent.com/1902116/61691286-80589a00-ad2b-11e9-8fad-94c025f4bf04.png">

**With Solution Values checked, the concentration is actually 1.000 molar**
<img width="842" alt="Screen Shot 2019-07-23 at 9 16 11 AM" src="https://user-images.githubusercontent.com/1902116/61691346-a41be000-ad2b-11e9-8dd7-4b4fe712d72c.png">

**Actually, zero concentration is showing up for several values past zero**
Please make sure the alert about concentration is synched with the model. It seems out of synch.
<img width="630" alt="Screen Shot 2019-07-23 at 9 35 10 AM" src="https://user-images.githubusercontent.com/1902116/61692220-864f7a80-ad2d-11e9-91ff-78650f637f09.png">

**Comment by terracoda** [comment]
I see, it looks like the correct described concentration scale is implemented. All descriptions show up for potassium permangenate. 
<img width="670" alt="Screen Shot 2019-07-23 at 9 45 36 AM" src="https://user-images.githubusercontent.com/1902116/61692919-df6bde00-ad2e-11e9-9e05-407fde8d6f7c.png">
 
For some solutes, however, the ranges are wider and there is a range implemented for "zero concentration". "zero concentration" does not have a range - only a single value, and I don't think it actually shows up in alerts. When there is "zero concentration" the alert is "No solute. Solution is pure water."

"max concentration" has a range, but I don't think it should actually show up in the alerts, either. In the alerts,  when a solution becomes saturated (at max concentration), the alert is "Solution saturated. Beyond max concentration, solids appear.", and the concentration level in the PDOM would be "has max concentration".

When a solution becomes unsaturated, it is not at "max concentration", it should be "highly concentrated".
<img width="642" alt="Screen Shot 2019-07-23 at 10 07 59 AM" src="https://user-images.githubusercontent.com/1902116/61694832-961d8d80-ad32-11e9-99c2-215296e26cba.png">

**Comment by terracoda** [comment]
@twant, I have reviewed. 
The concentration levels are in, but the ranges and corresponding alerts need some work.
Is the issue clearly described above? 

Assigning back to you, let me know if you have questions.

**Comment by terracoda** [comment]
@twant and @Matthew-phet, I also just added  #134 which is somewhat related.

**Comment by twant** [comment]
@terracoda @Matthew-phet pushed changes to concentration scale to master. assigning back over to you and will send a dev version tonight once a few more issues have been addressed!

**Comment by terracoda** [comment]
@twant, with Solution Values not checked, I still see "Solution no longer saturated. Solution **has max concentration**, solids dissolve." in the alert for:
- orange solution, Potassium dichromate

All other solutions seem to have the correct alert, which I believe to be:
- "Solution no longer saturated. Solution **is highly concentrated**, solids dissolve." when saturation is lost.

There might be a small rounding issue with the saturation point for Potassium dichromate that might need to be tweaked? I defer to you and @Matthew-phet to discuss how to handle this.

With Solution Values checked there are some more nuanced issues with the alert phrasing which will be followed in at least one other issue #134.

**Comment by terracoda** [comment]
Forgot screenshot from master, here you go:
<img width="674" alt="Screen Shot 2019-08-05 at 11 10 26 AM" src="https://user-images.githubusercontent.com/1902116/62486355-e6a1db80-b773-11e9-933c-907574e60132.png">

**Comment by terracoda** [comment]
I repeat, this seems to only happen with Potassium dichromate. Working nicely for all other solutions.

**Comment by emily-phet** [comment]
@terracoda It's hard to tell what's happening here without knowing the exact value you're at. When I use the sim, (dev 22) I can't find this same readout. Can you reproduce and then use the show values checkbox to show exactly what value you're at?

**Comment by emily-phet** [comment]
@terracoda If you don't want or are unable to spend time on this soon, we can discuss together on Tuesday.

**Comment by terracoda** [comment]
@emily-phet, sure, that makes sense as I think it might be a rounding error.

**Comment by terracoda** [comment]
It only happens when Solute Amount is 0.500 moles and Solution Volume is 1.000 litres.
<img width="449" alt="Screen Shot 2019-08-08 at 4 44 29 PM" src="https://user-images.githubusercontent.com/1902116/62744642-588c5600-b9fc-11e9-9000-b17c1b1736af.png">

@emily-phet, is the alert content accurate? The solution's concentration is 0.500 Molar.

**Comment by terracoda** [comment]
I thought if the solution was no longer saturated, the alert would have to be:
"Solution no longer saturated. Solution highly concentrated, solids dissolve."

However, since it is actually at the saturation point, maybe the alert is correct.

**Comment by terracoda** [comment]
@emily-phet, this is actually a very nuanced situation, as a content expert would you describe the saturation point as, "Solution no longer saturated. Solution has max concentration, solids dissolve."

In my non-expert understanding, the solution is actually saturated, but not yet "beyond max concentration". 

Perhaps a more accurate alert for the rare occurrence of reaching the exact saturation point is some variant of the following:
- "More/Less solution, saturated. Solution at max concentration, all solids dissolved."
- "More/Less solute, saturated. Solution at max concentration, all solids dissolved."

**Comment by Matthew-phet** [comment]
@terracoda I think its only considered saturated if the sim itself says that its saturated

**Comment by terracoda** [comment]
@Matthew-phet, sorry, my question is, what should the alert be when the solution actually hits the exact saturation point, i.e. the solution "has max concentration" and there are no solids.

I think we have a good alerts for landing past the saturation point or landing below the saturation point. 

What's the best/most accurate way to describe hitting the exact saturation point? 

Do you think, "Solution no longer saturated. Solution has max concentration (i.e., 0.500 molar), solids dissolve." correctly describes hitting the saturation point?

More specifically, can it be true that the solution is "no longer saturated" and also has "max concentration"?

**Solution Values not checked:**
<img width="719" alt="Screen Shot 2019-08-09 at 8 47 16 AM" src="https://user-images.githubusercontent.com/1902116/62792053-499cb680-ba83-11e9-885b-ffdda86de035.png">

**Same values with Solution Values checked:**
<img width="715" alt="Screen Shot 2019-08-09 at 8 47 41 AM" src="https://user-images.githubusercontent.com/1902116/62792136-7f419f80-ba83-11e9-83ef-3ad04c3b1310.png">

**Comment by terracoda** [comment]
@Matthew-phet and @emily-phet, I think I have clarified the question I am asking. I'm adding the meeting label, so we are sure to discuss this on Tuesday.

**Comment by Matthew-phet** [comment]
Heres the link for the [agenda](https://docs.google.com/document/d/105AnA7Z-TbwIGTHyHQCxK-ZNc9LwfjsQx9UtmMBnTHc/edit#) it should have all the info needed.

**Comment by terracoda** [comment]
@twant, my understanding is that you and @Matthew-phet discussed the changes documented at the top of [Molarity's Agenda doc](https://docs.google.com/document/d/105AnA7Z-TbwIGTHyHQCxK-ZNc9LwfjsQx9UtmMBnTHc/edit#)

After your meeting yesterday, I noticed 2 small issues:
1. Example 2c - no need to use "at"
2. Example 4c - need to use the word "Concentration" instead of "Solution"

Hope all is clear. Please reach out if there are any questions.

**Comment by twant** [comment]
Okay, I think I've gotten to the bottom of this one and pushed to master. Assigning over to you for review, @Matthew-phet and @terracoda! I have one more issue to finish and then I'll create a dev version as well.

**Comment by terracoda** [comment]
@Matthew-phet, I am leaving this one for you to verify. Unassigning myself for now.

**Comment by twant** [comment]
- [ ] Need to remove double period on initial alert

- [ ] Make the final value before saturation give a region alert at the end of it (for 2c)

**Comment by twant** [comment]
@Matthew-phet and @terracoda I fixed this one so that "solution has max concentration" appears just before saturation in the alerts. However, for some combinations volume and solute amount, increasing solute amount increases concentration by increments larger than .001 (even with the shift key down), so it skips this single-value region. Is this okay with you?

**Comment by terracoda** [comment]
@twant, I think this is fixed. 
The word "concentration" is in the alert when solution values i checked.
The ranges for the described levels seem to work well when solution values is not checked.

@twant, about skipping the single-value region "has max concentration", I think that's the nature of the beast in the approximated described regions. This also happens with values checked. The actual saturation point in molar is often not communicated. The alert is:
- "Solution saturated. Beyond max concentration, solids appear."

A student would have to investigate the Beaker description to get the exact value for molar for some solutions.
 
I think it is safe to close this one.
